tests/interfaces/test_batch_jax_qnode.py

```diff
@@ -486,32 +515,101 @@ def circuit(x, y):
             ]
         )
         assert np.allclose(res, expected, atol=tol, rtol=0)
+        if diff_method in ("parameter-shift", "finite-diff"):
 
-        res = jax.jacobian(circuit, argnums=[0, 1])(x, y)
+            res = jax.jacobian(circuit, argnums=[0, 1])(x, y)
+
+            # TODO: remove the swapping of axes when custom gradient outputs
+            # have been adjusted

```

Comment:
What do you mean by this?

Reply:
Nice! That's awesome @antalszava.

One thing maybe to double check: _should_ the JAX interface be matching the other interfaces?

We can double check this by comparing the interfaces without PL:

```python
def cost(x, y):
    ...
    return tensor1, tensor2
```

If JAX disagrees with torch/tf/autograd in this case, maybe we should preserve this when used with PL? 🤔

Reply:
Torch and JAX seem to match:
```python
import jax
from jax import numpy as jnp

fun = lambda x, y: tuple([x**2, y**3])
a = jnp.array([1.,4.])

print('JAX: \n', jax.jacobian(fun, argnums=[0,1])(a[0], a[1]))

import torch

a = torch.tensor([1.,4.])
inputs = (a[0], a[1])

fun = lambda x, y: tuple([x**2, y**3])
print('\nTorch: \n', torch.autograd.functional.jacobian(fun, inputs))
```
```
JAX: 
 ((DeviceArray(2., dtype=float32), DeviceArray(0., dtype=float32)), (DeviceArray(0., dtype=float32), DeviceArray(48., dtype=float32)))

Torch: 
 ((tensor(2.), tensor(0.)), (tensor(0.), tensor(48.)))
```
Couldn't get autograd and TF to work when computing the jacobian of functions returning a tuple.

Reply:
Nice, thanks for checking this Antal!